### PR TITLE
Added Dynamic Deference & Allocation WG

### PR DESCRIPTION
**Pull request by jwoehr**
### Summary
Add new Working Group description

### Details and comments
Voted by the TSZC 2022-01-28

### CODE REVIEW

**Comment by jwoehr** [comment]
Voted by TSC 2022-01-28

File: WG.md [hunk 1]
```diff
@@ -12,12 +12,12 @@ for review by the TSC at an agreed upon date. A WG will be automatically disband
...
 
-Chair: Niel de Beaudrap  
+Chair: Niel de Beaudrap
 Members: Colm Ryan, Andrew Cross, Ali Javadi, Blake Johnson, Matthew Amy
 
 ### Types and casting
 
-**Objective**: Define type hierarchy and implicit casting rules.  
+**Objective**: Define type hierarchy and implicit casting rules.
 **Questions**:
```
Comment (1ucian0) [inline WG.md:21]:
```suggestion
**Objective**: Define type hierarchy and implicit casting rules.  
**Questions**:
```

File: WG.md [hunk 2]
```diff
@@ -12,12 +12,12 @@ for review by the TSC at an agreed upon date. A WG will be automatically disband
 
 **Objective**: Recommend what range of generics / parameterised circuit family functionality (beyond angle parameters) would be suitable for inclusion in OpenQASM, and what syntax should be used for it.
 
-Chair: Niel de Beaudrap  
+Chair: Niel de Beaudrap
 Members: Colm Ryan, Andrew Cross, Ali Javadi, Blake Johnson, Matthew Amy
```
Comment (1ucian0) [inline WG.md:16]: These trail spaces create a breakline here. See https://gist.github.com/shaunlebron/746476e6e7a4d698b373

```suggestion
Chair: Niel de Beaudrap  
Members: Colm Ryan, Andrew Cross, Ali Javadi, Blake Johnson, Matthew Amy
```

File: WG.md [hunk 3]
```diff
@@ -26,13 +26,13 @@ Members: Colm Ryan, Andrew Cross, Ali Javadi, Blake Johnson, Matthew Amy
  * Can you index into integers and get bit value?
  * Is an int equivalent to an array of bits?
 
-Chair: Michael Healy (IBM Quantum)  
+Chair: Michael Healy (IBM Quantum)
 Members: Niel de Beaudrap, Hiroshi Horii, Blake Johnson, Colm Ryan, Luciano Bello, Prasahnt Sivarajah, Yunong Shi, Jake Lishman
```
Comment (1ucian0) [inline WG.md:30]:
```suggestion
Chair: Michael Healy (IBM Quantum)  
Members: Niel de Beaudrap, Hiroshi Horii, Blake Johnson, Colm Ryan, Luciano Bello, Prasahnt Sivarajah, Yunong Shi, Jake Lishman
```

File: WG.md [hunk 4]
```diff
@@ -26,13 +26,13 @@ Members: Colm Ryan, Andrew Cross, Ali Javadi, Blake Johnson, Matthew Amy
...
 
 ### Pragmas
 
 
-**Objectives**: Define a pragma syntax, standard directives, the standard behavioral expectations for interacting within a toolchain.  
+**Objectives**: Define a pragma syntax, standard directives, the standard behavioral expectations for interacting within a toolchain.
 **Questions**:
```
Comment (1ucian0) [inline WG.md:36]:
```suggestion
**Objectives**: Define a pragma syntax, standard directives, the standard behavioral expectations for interacting within a toolchain.  
**Questions**:
```

**Review by 1ucian0 — CHANGES_REQUESTED** [review]
Those extra spaces at the end have a meaning :)

File: WG.md [hunk 5]
```diff
@@ -56,5 +71,5 @@ BẢO Bach Gia (Ho Chi Minh University of Technology), Jake Lishman (IBM Quantum
 **Objective**: Define a pulse grammar "openpulse" to be used for microcoding of gate instructions with
 OpenQASM `defcal`'s.
 
-Chair: Thomas Alexander (IBM Quantum)  
+Chair: Thomas Alexander (IBM Quantum)
 Members: Blake Johnson, Colm Ryan, Derek Bolt, Peter Karalekas, Lauren Capelluto, Michael Healy, Prasahnt Sivarajah, Yunong Shi, Steven Heidel
```
Comment (1ucian0) [inline WG.md:75]:
```suggestion
Chair: Thomas Alexander (IBM Quantum)  
Members: Blake Johnson, Colm Ryan, Derek Bolt, Peter Karalekas, Lauren Capelluto, Michael Healy, Prasahnt Sivarajah, Yunong Shi, Steven Heidel
```

File: WG.md [hunk 6]
```diff
@@ -43,11 +43,26 @@ Members: Niel de Beaudrap, Hiroshi Horii, Blake Johnson, Colm Ryan, Luciano Bell
...
 
-Chair: Jeff Heckey (AWS)  
+Chair: Jeff Heckey (AWS)
 Members: Ali Javadi (IBM Quantum)
 BẢO Bach Gia (Ho Chi Minh University of Technology), Jake Lishman (IBM Quantum), Jack Woehr (IBM Quantum), Lev S. Bishop (IBM Quantum), Michael Healy (IBM Quantum), Thomas Alexander (IBM Quantum), Yunong Shi (AWS)
 
+### Dynamic Deference & Allocation
 
+**Objectives**: Formulate response to issues raised by [Physical mapping of dynamically dereferenced qubits #307](https://github.com/Qiskit/openqasm/issues/307)
+ * Provide oncrete examples proving this cannot be dealt with in OpenQASM 3 as the spec stands
```
Comment (jakelishman) [inline WG.md:53]:
```suggestion
 * Provide concrete examples proving this cannot be dealt with in OpenQASM 3 as the spec stands
```

**Review by 1ucian0 — APPROVED** [review]

**Pull request merged**